js/web/lib/wasm/jsep/webgpu/ops/transpose.ts

```diff
@@ -23,8 +23,9 @@ const validateInputs = (inputs: readonly TensorView[]): void => {
     throw new Error('Transpose requires 1 input.');
   }
 
-  if (inputs[0].dataType !== DataType.float) {
-    throw new Error('input should be float tensor');
+  if (inputs[0].dataType !== DataType.float && inputs[0].dataType !== DataType.int32 &&
```

Comment:
nit: don't think this checkk is needed since the registration already takes care of it.

Reply:
I thought so too - but kept it consistent with how it was previously.